### PR TITLE
Add schedulerName to selector list

### LINKED ISSUE #48453: schedulerName field not supported in fieldselector
/kind bug
/sig api-machinery

**What happened**: trying to select pods based on which scheduler they are using

**What you expected to happen**: using fieldSelector would enable me to defer the filtering to the api server instead of using a local for loop over all pods

**How to reproduce it (as minimally and precisely as possible)**:
```
$ curl https://kubernetes.default.svc/api/v1/pods?fieldSelector=spec.schedulerName%3Ddefault -k -H 'Authorization: Bearer REDACTED'
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {},
  "status": "Failure",
  "message": "field label not supported: spec.schedulerName",
  "reason": "BadRequest",
  "code": 400
}
```

**Anything else we need to know?**:

**Environment**: minikube
```
Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.0", GitCommit:"d3ada0119e776222f11ec7945e6d860061339aad", GitTreeState:"clean", BuildDate:"2017-06-30T09:50:48Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.4", GitCommit:"d6f433224538d4f9ca2f7ae19b252e6fcb66a3ae", GitTreeState:"dirty", BuildDate:"2017-06-22T04:31:09Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
```
In general I would expect fieldSelectors to work for all fields. I understand it's a discouraged feature, but there really isn't any other way to do efficient filtering without just recreating the entire spec in the labels. I found other fields that also don't work, but I am most interested in the newly added schedulerName because I run a custom priority scheduler in my clusters.

### PR DESCRIPTION
**Pull request by mkumatag**
**What this PR does / why we need it**:
Adds schedulerName to selector list
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: 
fixes #48453 

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by mkumatag** [comment]
fixed the go fmt stuff!

**Review by gyliu513 — COMMENTED** [review]
/lgtm

**Comment by k82cn** [comment]
Do we have any document on which fields are available? If not, suggest to create one. @@lavalamp, WDYT?

File: pkg/api/v1/conversion.go
Comment (k82cn) [inline pkg/api/v1/conversion.go]: not this PR :). But when `spec.host` can be removed? If not sure, suggest to depreciated since 1.8;  and remove it in 1.10 (?).

**Comment by k82cn** [comment]
thanks very much for the PR; that's great, we can also update default-scheduler to use `schedulerName` for multi-scheduler after this PR merged.

/cc @davidopp @timothysc  @bsalamat

**Comment by mkumatag** [comment]
@k82cn I don't see any proper doc at the moment, I think we can update swagger-spec to update api-reference guide. WDYT?
> Do we have any document on which fields are available? If not, suggest to create one. @@lavalamp, WDYT?

**Comment by k82cn** [comment]
Sorry for /cc so many people, I means F.Y.I, not review request in this context :).

**Comment by mkumatag** [comment]
I tried enhancing the document in https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L312 and resulted in generating lot of swagger update wherever this field is used.

Can someone help me find a place where I can put this information to enhance the api reference guide which talks about all possible `fieldSelector` values

**Comment by k82cn** [comment]
/retest

@mkumatag , I think we can add some doc similar to https://kubernetes.io/docs/reference/labels-annotations-taints/ . @lavalamp, @smarterclayton , any comments?

File: pkg/registry/core/pod/strategy_test.go
Comment (timothysc) [inline pkg/registry/core/pod/strategy_test.go]: Why is this "custom-scheduler" here?

Comment (mkumatag) [inline pkg/registry/core/pod/strategy_test.go]: This is just a negative test and trying to list which does not exist, this value can be anything in fact.

Comment (timothysc) [inline pkg/registry/core/pod/strategy_test.go]: Sorry, missed the false below.

**Review by timothysc — APPROVED** [review]
/lgtm - but needs registry approver.

**Comment by mkumatag** [comment]
ping @lavalamp 
/retest

**Comment by liggitt** [comment]
/approve
needs rebase and release note

File: pkg/api/v1/conversion.go
Comment (liggitt) [reply pkg/api/v1/conversion.go]: I wouldn't expect it to be removed

**Comment by liggitt** [comment]
just noticed this was tagged for 1.8... needs an exception request as well

**Comment by mkumatag** [comment]
I think the code is already implemented and merged part of https://github.com/kubernetes/kubernetes/pull/50582, hence closing this PR.

**Pull request closed without merging**
